dask/delayed.py

```diff
@@ -4,6 +4,7 @@
 import warnings
 from collections.abc import Iterator
 from dataclasses import fields, is_dataclass
+from typing import Any, Hashable, Optional
```

Comment:
You should import from collections.abc when possible. Refer to Python 3.9 release notes for the rationale.